EVG-7415: Query Patch Code Changes

SupaJoon opened the pull request:
https://jira.mongodb.org/browse/EVG-7415
This PR includes the `moduleCodeChanges` field on the Patch GQL type. 
The schema updates support the Changes tab in the Patch Page.
<img width="1171" alt="Screen Shot 2020-03-20 at 6 18 49 PM" src="https://user-images.githubusercontent.com/10734386/77210454-6d26e580-6ad7-11ea-8090-cf50c2110543.png">

Next I'll develop the frontend code for the screenshot above ^

john-m-liu commented on rest/model/patch.go:
```diff
@@ -11,30 +14,45 @@ import (
+	BranchName *string     `json:"branch_name"`
+	HTMLLink   *string     `json:"html_link"`
+	RawLink    *string     `json:"raw_link"`
+	FileDiffs  []*FileDiff `json:"file_diffs"`
```
Since you declared the contents of this array as non-nullable in the schema, this doesn't need to be a pointer does it? In general we tend to avoid arrays of pointers because they have some quirky behavior when you iterate over them

john-m-liu commented on graphql/schema.graphql:
```diff
@@ -100,6 +114,7 @@ type Patch {
   duration: PatchDuration
   time: PatchTime
   taskCount: Int
+  moduleCodeChanges: [ModuleCodeChange]
```
This will probably be easier to use if the contents are non-nullable. Also potentially the field itself as well (it is valid to have a patch with no code changes)

tgrander commented on rest/model/patch.go:
```diff
@@ -11,30 +14,45 @@ import (
+	Activated         bool              `json:"activated"`
+	Alias             *string           `json:"alias,omitempty"`
+	GithubPatchData   githubPatch       `json:"github_patch_data,omitempty"`
+	ModuleCodeChanges []*APIModulePatch `json:"module_code_changes"`
```
why is `APIModulePatch` a pointer?

SupaJoon replied:
APIModulePatch` shouldn't be a pointer anymore since I'm making the moduleCodeChanges gql field's contents in type Task non-nullable

SupaJoon commented:
@john-m-liu Since you last reviewed these code changes, I found a bug that made all the fileNames the same. I fixed this here: https://github.com/evergreen-ci/evergreen/pull/3328/commits/f78506af478c91f4c745c8fc03dd1edb281a641e#diff-fe7723ff85e576d55caf6778a1020713R112.
I also added in test for this as well as additional testdata for cypress integration testing
